src/recombinations.jl

```diff
@@ -481,6 +481,28 @@ function POS(v1::T, v2::T; rng::AbstractRNG=Random.GLOBAL_RNG) where {T <: Abstr
     return c1,c2
 end
 
+# Combinatorial crossovers
+# ----------------------
+
+"""
+    SXO(v1, v2)
+
+Subset crossover[^7]. Produces two offsprings by first pooling the unique items
+of the two parents, and then creating each offspring by sampling without
+replacement from the pool of items.
+"""
+function SXO(v1::T, v2::T; rng::AbstractRNG=Random.GLOBAL_RNG) where {T <: AbstractVector{Bool}}
+    l = length(v1) # get number of available elements
+    K = sum(v1) # get required subset size
+    @assert (sum(v2)==K) "v1 and v2 must have the same number of true entries"
```

Comment:
I do not think it's a correct implementation. The paper states:
> Each new offspring is formed by first pooling the unique indices of the two parent vectors, and then sampling a set of k unique indices from this pool, uniformly at random.

In the R package, `k` is a global parameters to the optimization, and it's independent from the particular individual. You calculates `k` as `sum(v1)` which is a number of `1`s in the binary individual. Thus, the `k` parameter will be different for each individual in that case. I am not sure how it will affect the course of the optimization, but it looks like the more `1`s appear in the genome, the more shuffled offspring will become.

The assert is also problematic. You cannot expect that individuals will have the same number of `1`s in their genome.

Reply:
I think now I see your point. I was thinking of `SXO` for the particular use case of k-subset optimization where all genetic operations maintain the subset size; i.e., sum(v1)==sum(v2)==k and is invariant across iterations.

But you (basically) say: "hey this recombination could be used in different contexts." And in those context, sum(v1) and sum(v2) may not coincide and moreover, neither may be equal to the target k. I think this is a good point and I'll modify `SXO` to reflect it, using the closure approach.